command/genesis/genesis.go

```diff
@@ -25,8 +25,6 @@ func GetCommand() *cobra.Command {
 	setFlags(genesisCmd)
 	setLegacyFlags(genesisCmd)
 
-	helper.SetRequiredFlags(genesisCmd, params.getRequiredFlags())
-
```

Comment:
Why was this moved to `runPreRun`?

Please move it back, as all methods that have required flags initialize them in the standard `GetCommand` method

Reply:
This is not the correct place in init flow, in this case, different required flags are required depending on the selected consensus, at this point of the execution `if p.isIBFTConsensus()` or `p.consensusRaw` is set to its default value.

Conditionally required flags should be setup in the `PreRunE`, see for example https://github.com/spf13/cobra/issues/1595#issuecomment-1046681041